Fix calculation of relative path for symlink on docker save

dmcgowan opened the pull request:
Relative paths are now calculated from a base path rather than from the file path, which gets treated as a directory. Symlinks will now properly point to the file as `../<layer dir>/layer.tar` rather the incorrect `../../<layer dir>/layer.tar`.

Fixes #24951

thaJeztah commented:
ping @aaronlehmann ptal

aaronlehmann commented on image/tarexport/save.go:
```diff
@@ -307,7 +307,7 @@ func (s *saveSession) saveLayer(id layer.ChainID, legacyImg image.V1Image, creat
 
 	if oldPath, exists := s.diffIDPaths[l.DiffID()]; exists {
-		relPath, err := filepath.Rel(layerPath, oldPath)
+		relPath, err := filepath.Rel(filepath.Dir(layerPath), oldPath)
```
Since `layerPath` is defined with `layerPath := filepath.Join(outDir, legacyLayerFileName)`, shouldn't this be `filepath.Rel(outDir, oldPath)`?

dmcgowan replied:
Updated

dmcgowan committed:
Fix calculation of relative path for symlink on docker save

Relative paths are now calculated from a base path rather than from the file path, which gets treated like a directory.
Symlinks will now properly point to the file as "../<layer dir>/layer.tar" rather the incorrect "../../<layer dir>/layer.tar".

Fixes #24951

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

aaronlehmann commented:
LGTM

thaJeztah commented:
LGTM
